components/cmake/cmake_util.cmake

```diff
@@ -66,10 +66,22 @@ function(e3sm_remove_flags FILE_ARG FLAGS_ARG)
     if (NOT EXISTS ${PROJECT_SOURCE_DIR}/${FILE_ARG})
       message(FATAL_ERROR "Trying to set flags on non-existent source: ${FILE_ARG}")
     endif()
-    set(REAL_FILE "../../${FILE_ARG}")
+    set(REAL_FILE "${SOURCE_PATH}/${FILE_ARG}")
   endif()
 
   get_property(ITEM_FLAGS SOURCE ${REAL_FILE} PROPERTY COMPILE_FLAGS)
   string(REPLACE "${FLAGS_ARG}" "" ITEM_FLAGS "${ITEM_FLAGS}")
   set_property(SOURCE ${REAL_FILE} PROPERTY COMPILE_FLAGS "${ITEM_FLAGS}")
 endfunction()
+
+function(e3sm_deoptimize_file FILE_ARG FFLAGS_NOOPT)
+  if (CMAKE_Fortran_COMPILER_ID STREQUAL "PGI")
+    # PGI does not support bulk-disabling of optimization by appending -O0,
+    # we have to remove the optimization flags first
+
+    # Until we know which particular flags are related to optimization, we have to guess.
+    e3sm_remove_flags(${FILE_ARG} "-O2")
+    e3sm_remove_flags(${FILE_ARG} "-O3")
```

Comment:
Should remove `-O1` too?

Reply:
Maybe, since CIME does not currently distinguish between optimization flags and other flags, I'm kind of guessing here.

Reply:
My vote is to remove `-O1` too, because pgi's `FFLAGS_NOOPT` is set to `-O0`.

Reply:
@amametjanov , will do